docs-configuration.md

```diff
@@ -0,0 +1,352 @@
+
+
+```python
+config = Config(name='dev',
+                properties={
+                   'my_project': 'I_am_{env}'
+                }).add_configuration(
+                name='prod',
+                properties={})
+
+config.pretty_print()
+config.pretty_print('dev')
+
+```
+
+output:
+
+```text
+dev config:
+{'my_project': 'I_am_dev'}
+dev config:
+{'my_project': 'I_am_dev'}
+```
+
+You can change the roles by setting the `is_default` flag:
+
+```python
+config = Config(name='dev',
+                is_default=False,
+                properties={
+                   'my_project': 'I_am_{env}'
+                }).add_configuration(
+                name='prod',
+                is_default=True,
+                properties={})
+
+config.pretty_print()
+config.pretty_print('dev')
+
+```
+
+output:
+
+```text
+prod config:
+{'my_project': 'I_am_prod'}
+dev config:
+{'my_project': 'I_am_dev'}
+
+### Operating System environment variables support
+
+**Secrets** shouldn't be stored in source code, but still, you need them in your configuration.
+Typically, they are stored safely on CI/CD servers and they are 
+passed around as Operating System environment variables.
+
+`Config` object supports that approach. When a property is not defined explicitly in Python,
+it is resolved from an OS environment variable. For example:
+
+
+```python
+import os
+from bigflow import Config
+
+os.environ['bgq_my_secret'] = '123456'
+
+config = Config(name='dev',
+                properties={
+                   'my_project': 'I_am_{env}'
+                })
+
+config.pretty_print('dev')
+print ('my_secret:', config.resolve_property('my_secret', 'dev'))
+```
+
+output:
+
+```text
+dev config:
+{'my_project': 'I_am_dev'}
+my_secret: 123456
+```
+
+There are two important aspects here.
+**First**, by default, OS environment variable names
+must be prefixed with `bgq_`. You can change this prefix by setting
+the `environment_variables_prefix` parameter in the `Config` init method.
+
+**Second**, since secret properties don't exist in Python code
+they are resolved always lazily and only by name, using the `Config.resolve_property()` function.
+
+
+More interestingly, you can even resolve a current configuration name from OS environment:
+
+```python
+import os
+from bigflow import Config
+
+config = Config(name='dev',
+                properties={
+                   'my_project': 'I_am_{env}'
+                }).add_configuration(
+                name='prod',
+                properties={})
+
+os.environ['bgq_env'] = 'prod'
+config.pretty_print('')
+os.environ['bgq_env'] = 'dev'
+config.pretty_print('')
+```
+
+output:
+
+```text
+prod config:
+{'my_project': 'I_am_prod'}
+dev config:
+{'my_project': 'I_am_dev'}
+```
+ 
+## Dataset Config
+
+`DatasetConfig` is a convenient extension to `Config` designed for workflows which use `DatasetManager`
+to call Big Query SQL. 
+
+`DatasetConfig` defines four properties that are required by `DatasetManager`:
```

Comment:
- Can I define some custom property and use it in a table name? 
- Can I use `{env}` in a table name?

Reply:
No, string interpolation works only for atomic String props